File: lua/productshotai_site_kit.lua
local M = {}
M.BASE = "https://productshotai.app"
M.BRAND = "ProductShot AI"
M.DESCRIPTION = "AI product photography generator for ecommerce sellers."

local function page_url(path)
  local value = path or "/"
  if value == "" then value = "/" end
  if value:sub(1, 1) ~= "/" then value = "/" .. value end
  local clean = value == "/" and "/" or value:gsub("/+$", "") .. "/"
  return M.BASE .. clean
end

function M.localized_url(locale, path)
  if locale == "en" then return page_url(path) end
  if locale == "zh" or locale == "zh-CN" then
    local value = path or "/"
    if value == "" then value = "/" end
    if value:sub(1, 1) ~= "/" then value = "/" .. value end
    return page_url("/zh" .. (value == "/" and "" or value))
  end
  error("unsupported locale: " .. tostring(locale))
end

function M.metadata()
  return {
    name = M.BRAND, homepage = M.BASE, description = M.DESCRIPTION,
    canonical_pages = { home = M.home_url(), workbench = M.workbench_url(), pricing = M.pricing_url(), blog = M.blog_url(), contact = M.contact_url(), zh_home = M.zh_home_url() },
    tags = { "productshot", "ai-product-photography", "ecommerce-product-photos", "url-helpers" }
  }
end

function M.home_url() return page_url("/") end
function M.workbench_url() return M.BASE .. "/#workbench" end
function M.pricing_url() return M.BASE .. "/#pricing" end
function M.blog_url() return page_url("/blog") end
function M.about_url() return page_url("/about") end
function M.contact_url() return page_url("/contact") end
function M.privacy_url() return page_url("/privacy") end
function M.terms_url() return page_url("/terms") end
function M.refund_policy_url() return page_url("/refund-policy") end
function M.zh_home_url() return M.localized_url("zh", "/") end

return M

File: lib/productshotai_site_kit.ex
defmodule ProductshotaiSiteKit do
  @moduledoc "Public URL helpers for ProductShot AI."

  @base "https://productshotai.app"
  @brand "ProductShot AI"
  @description "AI product photography generator for ecommerce sellers."

  def base, do: @base
  def brand, do: @brand
  def description, do: @description

  def page_url(path \\ "/") do
    value = if path in [nil, ""], do: "/", else: path
    value = if String.starts_with?(value, "/"), do: value, else: "/#{value}"
    clean = if value == "/", do: "/", else: String.trim_trailing(value, "/") <> "/"
    @base <> clean
  end

  def localized_url("en", path), do: page_url(path)
  def localized_url(locale, path) when locale in ["zh", "zh-CN"] do
    value = if path in [nil, "", "/"], do: "/", else: path
    value = if String.starts_with?(value, "/"), do: value, else: "/#{value}"
    page_url("/zh" <> if(value == "/", do: "", else: value))
  end
  def localized_url(locale, _path), do: raise(ArgumentError, "unsupported locale: #{locale}")

  def metadata do
    %{
      name: @brand, homepage: @base, description: @description,
      canonical_pages: %{home: home_url(), workbench: workbench_url(), pricing: pricing_url(), blog: blog_url(), contact: contact_url(), zh_home: zh_home_url()},
      tags: ["productshot", "ai-product-photography", "ecommerce-product-photos", "url-helpers"]
    }
  end

  def home_url, do: page_url("/")
  def workbench_url, do: "#{@base}/#workbench"
  def pricing_url, do: "#{@base}/#pricing"
  def blog_url, do: page_url("/blog")
  def about_url, do: page_url("/about")
  def contact_url, do: page_url("/contact")
  def privacy_url, do: page_url("/privacy")
  def terms_url, do: page_url("/terms")
  def refund_policy_url, do: page_url("/refund-policy")
  def zh_home_url, do: localized_url("zh", "/")
end
